src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -1,43 +1,45 @@
 package edu.kis.vh.nursery;
 
-public class DefaultCountingOutRhymer {
+import edu.kis.vh.nursery.dataStructures.IntArrayStack;
+import edu.kis.vh.nursery.dataStructures.IntDataStructure;
+import edu.kis.vh.nursery.dataStructures.IntLinkedList;
 
-	private static final int INITIAL = -1;
-	private static final int MAX_SIZE = 12;
-	private static final int DEFAULT_VALUE = -1;
-	private static final int FULL = MAX_SIZE - 1;
+public class DefaultCountingOutRhymer {
 
-	private final int[] numbers = new int[MAX_SIZE];
+	private IntDataStructure intArrayStack = new IntArrayStack();
+	private IntDataStructure intLinkedList = new IntLinkedList();
```

Comment:
there should be a single IntDataStructure field!
intLinkedList is completely inactive!
The goal of the lab was to make using IntLinkedList and IntArrayStack interchangeable.